content/2016.02.localisation-service.md

```diff
@@ -0,0 +1,316 @@
+      "name": "Loop-Client translation",
+      "config": {
+        "displayFields": ["key", "value"],
+        "schema": {
+          "title": "Loop-Client translation",
+          "description": "Loop-Client translation strings",
+          "type": "object",
+          "additionalProperties": false,
+          "required": [
+            "key", "value"
+          ],
+          "properties": {
+            "key": {
+              "type": "string",
+              "title": "Key",
+              "description": "The translation key",
+              "minLength": 1,
+              "default": ""
+            },
+            "value": {
+              "type": "string",
+              "title": "Value",
+              "description": "The translation",
+              "minLength": 1,
+              "default": ""
+            }
+          }
+        }
+      }
+    }
+```
+
+Which generates the following simple form:
+
+![The Kinto Admin on the french category]({filename}/images/french-translation.png)
+
+When records are saved and synced on the server, the clients will be able fetch them.
+
+
+Synchronize and read translated strings with Kinto.js
+-----------------------------------------------------
+
+Load kinto.js component:
+
+```javascript
+    Cu.import("resource://services-common/moz-kinto-client.js");
+    const Kinto = loadKinto();
+```
+
+Synchronize from server on startup:
+
+
+Hack l10n.js to rely on Kinto for reading translated strings:
+
+```javascript
+    var gLocalizedStrings = new Map();
+
+    var MozLoopServiceInternal = {
+      // ...
+
+      configureGLocalizedStringsWithKinto: function() {
+        let FirefoxAdapter = Kinto.adapters.FirefoxAdapter;
+        let config = {
+           remote: "https://kinto-ota.dev.mozaws.net/v1",
+           bucket: "loop-client",
+           adapter: FirefoxAdapter
+        };
+        let db = new Kinto(config);
+        let collectionName = "fr"; // XXX: this.locale getter can help
+        let locales = db.collection(collectionName);
+  
+        return locales.db.open()
+          .then(() => {
+            return locales.sync();
+          })
+          .then(() => {
+            return locales.list();
+          })
+          .then((res) => {
+            res.data.forEach((record) => {
+              gLocalizedStrings.set(record.key, record.value);
+            });
+          })
+          .then(() => {
+            return locales.db.close();
+          });
+        },
+
+        initialize: Task.async(function*() {
+          // ...
+
+          // Start l10n initilization
+          yield this.configureGLocalizedStringsWithKinto();
+
+          // ...
+        },
+      // ...
+    }
+```
+
+## Demo
+
+{% video https://kinto-ota.dev.mozaws.net/attachments/demo.mp4 900 550 %}
+
+
+# Conclusion
+
+One of our objectives with this experiment was to illustrate what can
+be done with Kinto within Firefox.
+
+We have seen that using the embedded Kinto.js in Firefox is very
+simple, and playful.
```

Comment:
suggestion: ... is rather straightforward and delivers impressive results quickly as we implemented this proof of concept in just a few hours.